### ISSUE TITLE
No matching signature found

### ISSUE THREAD

**Issue by MatteoLacki** [calvin-sykes/cython_parallel_sort#2]
Hello,

I have a problem installing your package:

![image](https://github.com/calvin-sykes/cython_parallel_sort/assets/3964160/f505f983-6d19-4188-899e-a03e6c972e8e)

Have g++ 11.4.0 installd on ubuntu22.05LTS

Any ideas?

Best wishes,
Matteo

**Comment by calvin-sykes** [comment]
This package only works for sorting NumPy arrays, not python lists.

Thanks for the report though, I'll release a new version with a more helpful error message.

**Comment by MatteoLacki** [comment]
Hello!

Sorry, didn't notice. With most `numpy` arrays it works:
this works:
```
In [174]: parallel_sort.sort(np.array([3,2,1]))
Out[174]: array([1, 2, 3])
```

This works:
```
In [175]: parallel_sort.sort(np.array([3,2,1], dtype=np.uint32))
Out[175]: array([1, 2, 3], dtype=uint32)
```

This works:
```
In [187]: parallel_sort.sort(np.array([3.3,2.2,1.1]))
Out[187]: array([1.1, 2.2, 3.3])
```

This does work:
```
In [189]: parallel_sort.sort(np.array([3.3,2.2,1.1], dtype=np.float32))
Out[189]: array([1.1, 2.2, 3.3], dtype=float32)

In [190]: parallel_sort.sort(np.array([3.3,2.2,1.1], dtype=np.float64))
Out[190]: array([1.1, 2.2, 3.3])```
```

This does not run:
```
In [188]: parallel_sort.sort(np.array([3.3,2.2,1.1], dtype=np.half))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File parallel_sort/parallel_sort.pyx:34, in parallel_sort.check_ndarray.wrapped()

File parallel_sort/parallel_sort.pyx:44, in parallel_sort.__pyx_fused_cpdef()

TypeError: No matching signature found

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
Cell In[188], line 1
----> 1 parallel_sort.sort(np.array([3.3,2.2,1.1], dtype=np.half))

File parallel_sort/parallel_sort.pyx:36, in parallel_sort.check_ndarray.wrapped()

TypeError: Only NumPy arrays of arithmetic types may be sorted
```

Of course, likely not all people use `np.half`, but sometimes it's helping to stay within RAM bounds.

Best wishes!!

**Comment by calvin-sykes** [comment]
Float16 isn't a native C datatype, so I thought the only way to support it would be to convert to float32 before sorting, which obviously wouldn't be ideal for RAM usage. But it turns out numpy exposes a C API for working with float16, which includes sorting predicates. So I've released v0.2 in which this should now work.
